src/cryptonote_core/blockchain.cpp

```diff
@@ -3098,7 +3119,8 @@ bool Blockchain::check_fee(size_t tx_weight, uint64_t fee) const
   uint64_t needed_fee;
   if (version >= HF_VERSION_PER_BYTE_FEE)
   {
-    uint64_t fee_per_byte = get_dynamic_base_fee(base_reward, median, version);
+    const bool use_long_term_median_in_fee = version >= HF_VERSION_LONG_TERM_BLOCK_WEIGHT;
+    uint64_t fee_per_byte = get_dynamic_base_fee(base_reward, use_long_term_median_in_fee ? m_long_term_block_weights.back() : median, version);
```

Comment:
`m_long_term_block_weights` contains `LongTermBlockWeight` defined as `min(BlockWeight,1.4*LongTermEffectiveMedianBlockWeight)` which doesn't represent any sort of median; it's just the actual block weight being capped by some slowly moving number, so it can change quite abruptly over consecutive heights. Is this behavior intended?